File: Java/2 - WeatherStation (Observer)/src/main/java/CurrentConditionsDisplay.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */

/**
 *
 * @author ismas
 */
public class CurrentConditionsDisplay implements Observer, DisplayElement {
    private float temperature;
    private float humidity;
    private WeatherData weatherData;
    
    public CurrentConditionsDisplay( WeatherData weatherData ){
        this.weatherData = weatherData;
        weatherData.registerObserver(this);
    }
    
    public void update( float temperature, float humidity, float pressure ){
        this.temperature = temperature;
        this.humidity = humidity;
        display();
    }
    
    public void display(){
        System.out.println("Current conditions: " + temperature + "F degrees and " + humidity + "% humidity");
    }
}

File: C#/2. WeatherStation (Observer)/Classes/CurrentConditionsDisplay.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using WeatherStation.Interfaces;

namespace WeatherStation.Classes
{
    public class CurrentConditionsDisplay : IObserver
    {
        private float _Temperature;
        private float _Humidity;
        private WeatherData _WeatherData;

        public CurrentConditionsDisplay(WeatherData weatherData)
        {
            _WeatherData = weatherData;
            weatherData.RegisterObserver(this);
        }

        public void Update(float temperature, float humidity, float pressure)
        {
            _Temperature = temperature;
            _Humidity = humidity;
            Display();
        }

        public void Display()
        {
            Console.WriteLine("Current conditions: " + _Temperature + "F degrees and " + _Humidity + "% humidity");
        }
    }
}
